### PR TITLE
remove 6.4.x version constants

### PR DESCRIPTION
**Pull request by talevy**
relates refactoring initiative #41164.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/action/admin/indices/close/CloseIndexRequestTests.java [hunk 1]
```diff
@@ -54,7 +52,7 @@ public void testBwcSerialization() throws Exception {
         {
             final CloseIndexRequest request = randomRequest();
             try (BytesStreamOutput out = new BytesStreamOutput()) {
-                out.setVersion(randomVersionBetween(random(), Version.V_6_4_0, VersionUtils.getPreviousVersion(Version.V_7_2_0)));
+                out.setVersion(VersionUtils.randomCompatibleVersion(random(), Version.CURRENT));
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/action/admin/indices/close/CloseIndexRequestTests.java]: Should this be random version on or before 7.2? Seems by doing compatible version we have increased the possible versions it could be

File: server/src/test/java/org/elasticsearch/cluster/coordination/JoinTaskExecutorTests.java [hunk 2]
```diff
@@ -87,9 +87,10 @@ public void testPreventJoinClusterWithUnsupportedNodeVersions() {
             });
         }
 
-        Version oldMajor = Version.V_6_4_0.minimumCompatibilityVersion();
-        expectThrows(IllegalStateException.class, () -> JoinTaskExecutor.ensureMajorVersionBarrier(oldMajor, minNodeVersion));
-
+        if (minNodeVersion.onOrAfter(Version.V_8_0_0)) {
+            Version oldMajor = Version.V_7_2_0.minimumCompatibilityVersion();
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/cluster/coordination/JoinTaskExecutorTests.java]: Why is this changed to a hardcoded 7.2 compat version?

**Review by rjernst — APPROVED** [review]
Looks fine, just a question

File: server/src/test/java/org/elasticsearch/action/admin/indices/close/CloseIndexRequestTests.java (hunk 1 above)
Comment (talevy) [reply server/src/test/java/org/elasticsearch/action/admin/indices/close/CloseIndexRequestTests.java]: uh oh. you're right

File: server/src/test/java/org/elasticsearch/cluster/coordination/JoinTaskExecutorTests.java (hunk 2 above)
Comment (talevy) [reply server/src/test/java/org/elasticsearch/cluster/coordination/JoinTaskExecutorTests.java]: this was a mis-understanding of the code. Since it stopped things from failing, I thought it was right.

the reason it is confusing is because of how `minimumCompatibilityVersion` is calculated

**Comment by talevy** [comment]
run elasticsearch-ci/1

**Comment by talevy** [comment]
run elasticsearch-ci/1

**Comment by talevy** [comment]
run elasticsearch-ci/1
